samples/contrib/azure-samples/kfp-azure-databricks/databricks/_cluster_op.py

```diff
@@ -0,0 +1,274 @@
+            init_scripts: The configuration for storing init scripts.
+            spark_env_vars: An object containing a set of optional, user-specified environment
+                variable key-value pairs.
+            autotermination_minutes: Automatically terminates the cluster after it is inactive for
+                this time in minutes. If not set, this cluster will not be automatically terminated.
+            instance_pool_id: The optional ID of the instance pool to which the cluster belongs.
+
+        Raises:
+
+            ValueError: If no k8s resource name or Cluster name are provided.
+        """
+
+        if not spec:
+            spec = {}
+
+        if cluster_name:
+            spec["cluster_name"] = cluster_name
+        if num_workers:
+            spec["num_workers"] = num_workers
+        if autoscale:
+            spec["autoscale"] = autoscale
+        if spark_version:
+            spec["spark_version"] = spark_version
+        if spark_conf:
+            spec["spark_conf"] = spark_conf
+        if node_type_id:
+            spec["node_type_id"] = node_type_id
+        if driver_node_type_id:
+            spec["driver_node_type_id"] = driver_node_type_id
+        if custom_tags:
+            spec["custom_tags"] = custom_tags
+        if cluster_log_conf:
+            spec["cluster_log_conf"] = cluster_log_conf
+        if init_scripts:
+            spec["init_scripts"] = init_scripts
+        if spark_env_vars:
+            spec["spark_env_vars"] = spark_env_vars
+        if autotermination_minutes:
+            spec["autotermination_minutes"] = autotermination_minutes
+        if instance_pool_id:
+            spec["instance_pool_id"] = instance_pool_id
+
+        if not k8s_name and "cluster_name" in spec:
+            k8s_name = spec["cluster_name"]
+        elif not k8s_name:
+            raise ValueError("You need to provide a k8s_name or a cluster_name.")
+
+        super().__init__(
+            k8s_resource={
+                "apiVersion": "databricks.microsoft.com/v1alpha1",
+                "kind": "Dcluster",
+                "metadata": {
+                    "name": k8s_name,
+                },
+                "spec": spec,
+            },
+            action="create",
+            success_condition="status.cluster_info.state in (RUNNING, TERMINATED, UNKNOWN)",
+            attribute_outputs={
+                "name": "{.metadata.name}",
+                "cluster_id": "{.status.cluster_info.cluster_id}",
+                "cluster_name": "{.status.cluster_info.cluster_name}",
+                "state": "{.status.cluster_info.state}"
+            },
+            name=name)
+
+    @classmethod
+    def from_json_spec(cls,
+                       name: str = None,
+                       k8s_name: str = None,
+                       cluster_name: str = None,
+                       json_spec: str = None):
+        """Create a new instance of CreateClusterOp from a json specification.
+
+        Args:
+
```

Comment:
nit: redundant blank line

Reply:
I can remove the blank lines after e.g. "Args:", no problem. But when I do so, Visual Studio Code won't render the information for those Args properly. It will show all the args in a single line.

Reply:
Ummm... I'm okay with the blank line here. Have no experience with visual studio code.